### ISSUE TITLE
New tabs not shifted to left to expose closing 'X'

### ISSUE THREAD

**Issue by seattledeveloper** [Komodo/KomodoEdit#2227]
### Short Summary

When starting a new "Find in Files" search, the resulting tab doesn't automatically shift to the left to expose the closing 'X'.  I find this frustrating as I often click the X that closes those whole lower pane area rather than just the desire of closing the latest tab.  I first have to click the right arrow first to expose the X and then click the X.

### Steps to Reproduce

Open enough bottom tabs to fill the area, execute a new Find in Files search.

### Expected results

I believe the X should be exposed.

![bottom pane desired](https://cloud.githubusercontent.com/assets/9749337/21237828/7197ec72-c2b5-11e6-8b3c-8e3f1a1a8f03.png)

### Actual results

X is hidden.

![bottom pane default](https://cloud.githubusercontent.com/assets/9749337/21237853/8857d30a-c2b5-11e6-85c8-1915b60be8aa.png)

### Platform Information

Komodo IDE, version 10.1.3, build 89510, platform win32-x86.
Built on Thu Dec 01 10:26:16 2016.

### Additional Information

I also notice this same behavior in the tabs for files that are opened.

Existing behavior:

![top pane default](https://cloud.githubusercontent.com/assets/9749337/21237891/a9635e8e-c2b5-11e6-9674-1abb44a951b1.png)

Desired behavior:

![top pane desired](https://cloud.githubusercontent.com/assets/9749337/21237898/b095fb62-c2b5-11e6-8b33-2ab8a4ae41da.png)

Thanks in advance!

Ed

**Comment by Naatan** [comment]
This is just how tab scrolling works, you have to scroll the tab bar to reach the rest of it. If you often run into this I would recommend given the Open Files widget (left pane) a try.

**Comment by seattledeveloper** [comment]
I respectfully disagree -- and consider this to not be a good user experience (where the X to close a tab is covered by the X to close the whole window (pane).

Here's how two other programs deal with it that I did a quick search on.

First, this one opens new panes on the left (which I don't consider to be a solution worth pursuing, but just to point it out:

![textpad](https://cloud.githubusercontent.com/assets/9749337/21245548/e1f1d0f6-c2d7-11e6-8759-a8ac50faa479.png)

Here's a second one that shifts the tabs to the left when a new tab is opened (leaving a gap) and getting the tab away from the scroll arrows.  This seems like a worthy approach.

![homesite](https://cloud.githubusercontent.com/assets/9749337/21245560/0661fac4-c2d8-11e6-8dbf-a3754b5a8cf5.png)

Thoughts?

**Comment by cgchoffman** [comment]
@seattledeveloper, that first solution isn't really a solution.  If you go to the farthest right tab you'll hit the same problem I assume.  The second also seems like it would eventually fail once you have too many files open that it can't adjust tab widths.

Having said that, I do agree with @seattledeveloper that it's not unreasonable that we make sure that the entire current tab is showing which would bring the `x` into view. 

I tested this in the bottom pane with a **find results** tab that was hidden.  If I click on it, it is popped into full view which is what the file tab should do as well.

Here's a GIf of the above.
![a-tab-problem](https://cloud.githubusercontent.com/assets/655154/21245846/d6079a44-c2d9-11e6-8293-399399505c81.gif)

**Comment by seattledeveloper** [comment]
Just wanted to clarify -- the second solution just keeps moving the whole set of tabs to the left as new ones open, similar to clicking on the right arrow in Komodo.  What I'm suggesting as a solution is that when a new tab opens that isn't completely visible, that the app essentially fires off the click event on the right arrow to get the whole tab to be visible (hence not hiding the X on the tab).

**Comment by cgchoffman** [comment]
Oh sorry, I didn't read the whole issue.  I'm not actually able to reproduce this at all.  New tabs, whether a file tab or a find results tab, always open fully in view.

**Comment by seattledeveloper** [comment]
So on my machine, they are not in full view.  After my tab area is full, a new Find in Files tab opens like this:

![bottom pane default](https://cloud.githubusercontent.com/assets/9749337/21247190/7a9badea-c2e2-11e6-81a3-87144f6d9d93.png)

The X that is visible is the x that closes the whole bottom pane, not the one that just closes the Find "search for some ..." tab.  This is the essence of the request.

**Comment by cgchoffman** [comment]
> The X that is visible is the x that closes the whole bottom pane, not the one that just closes the Find "search for some ..." tab. This is the essence of the request.

Yep, I get it.  I'm saying it's not happening for me.  I wonder why our too machines behave differntly.
![a-tab-problem](https://cloud.githubusercontent.com/assets/655154/21247322/7caa9532-c2e3-11e6-955c-c992c86ca9b0.gif)

**Comment by seattledeveloper** [comment]
Interesting.  What I'm seeing from your video is what I'd expect to see, but not what I'm seeing.

I might have just uncovered the difference (this is subtle).  My machine operates just like yours *after* I've filled the tab area and gone on past the first one that fills the tab area.  Seems that it is the first tab that fills the area isn't being detected as filling the area, hence the no scrolling.  But once I create another tab, it then moves correctly, fully exposing the X on the tab.

It took about 5 tries to reproduce and find the odd behavior.  I might not have the reason right, but I am able to reliably reproduce both the behavior I am seeing and also the behavior you are seeing.

**Comment by cgchoffman** [comment]
> It took about 5 tries to reproduce and find the odd behavior.

Thanks for doing that digging.  It certainly narrows it down if @Naatan decides to open this.

**Comment by Naatan** [comment]
Thanks @cgchoffman, your gif better illustrated what the issue is to me.
